### PR TITLE
(maint) Refactor platform_data collection

### PR DESCRIPTION
**Pull request by mwaggett**
This commit updates platform_data to make use of tag_from_artifact_path when mapping platform tags to artifact paths for consumption by beaker. Previously, we were relying on artifacts_path for the mapping, which makes assumptions about what we expect to find on builds. Now we take the actual, existing artifact paths and map to tags.

### CODE REVIEW

**Comment by mwaggett** [comment]
see platform_data in the yaml here: http://builds.delivery.puppetlabs.net/puppet-agent/c1572b86b28872a61cd4b46d2281aa9dcffbdafc/artifacts/

**Comment by melissa** [comment]
It looks like this works
```
:platform_data:
  debian-8-i386:
    :artifact: "./debian-8-i386/puppet-agent_5.0.1.7.g405c49d-1jessie_i386.deb"
    :repo_config: "../repo_configs/deb/pl-puppet-agent-405c49d47f651724e3e6101276a795d894dd32d0-jessie.list"
  el-7-x86_64:
    :artifact: "./el-7-x86_64/puppet-agent-5.0.1.7.g405c49d-1.el7.x86_64.rpm"
    :repo_config: "../repo_configs/rpm/pl-puppet-agent-405c49d47f651724e3e6101276a795d894dd32d0-el-7-x86_64.repo"
  ubuntu-16.04-amd64:
    :artifact: "./ubuntu-16.04-amd64/puppet-agent_5.0.1.7.g405c49d-1xenial_amd64.deb"
    :repo_config: "../repo_configs/deb/pl-puppet-agent-405c49d47f651724e3e6101276a795d894dd32d0-xenial.list"
```

I still need to review the code though

**Comment by mwaggett** [comment]
\o/

File: lib/packaging/config.rb [hunk 1]
```diff
@@ -79,18 +79,15 @@ def platform_data
...
-              artifact = artifacts.find { |e| e.include? Pkg::Paths.artifacts_path(tag) and (e.include?("all") || e.include?("#{arch}.deb")) }
-              repo_config = "../repo_configs/deb/pl-#{self.project}-#{self.ref}-#{Pkg::Platforms.get_attribute(tag, :codename)}.list" if artifact
+              repo_config = "../repo_configs/deb/pl-#{self.project}-#{self.ref}-#{Pkg::Platforms.get_attribute(tag, :codename)}.list"
             when 'rpm'
-              artifact = artifacts.find { |e| e.include? Pkg::Paths.artifacts_path(tag) and e.include?(arch) }
-              repo_config = "../repo_configs/rpm/pl-#{self.project}-#{self.ref}-#{tag}.repo" if artifact
+              repo_config = "../repo_configs/rpm/pl-#{self.project}-#{self.ref}-#{tag}.repo"
             when 'swix', 'svr4', 'ips', 'dmg', 'msi'
-              artifact = artifacts.find { |e| e.include? Pkg::Paths.artifacts_path(tag) and e.include?(arch) }
             else
```
Comment (melissa) [inline lib/packaging/config.rb]: I'm not a huge fan of this empty block. Maybe just add a comment, like `# To nothing: there are no repo configs for these platforms`, or something? Whatever makes sense to you

**Comment by mwaggett** [comment]
comment added!

**Pull request merged**